source/util/LockFile.h

```diff
@@ -0,0 +1,36 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef AWS_IOT_DEVICE_CLIENT_LOCKFILE_H
+#define AWS_IOT_DEVICE_CLIENT_LOCKFILE_H
+
+#include <string>
+
+namespace Aws
+{
+    namespace Iot
+    {
+        namespace DeviceClient
+        {
+            namespace Util
+            {
+                struct LockFile
+                {
+                  private:
+                    static constexpr char TAG[] = "LockFile.cpp";
+                    std::string filename;
+
+                  public:
+                    explicit LockFile(const std::string &filename, const std::string &process);
```

Comment:
Can you add some documentation over here regarding this function and its use? You can check `StringUtils.h` for reference.

Reply:
sure